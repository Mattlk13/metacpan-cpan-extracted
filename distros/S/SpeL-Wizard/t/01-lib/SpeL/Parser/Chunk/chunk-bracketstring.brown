$tex = bless( {
  'path' => 't/01-lib/SpeL/Parser/Chunk',
  'tree' => bless( {
    'ElementList' => bless( {
      'Element' => [
        bless( {
          'MathUnit' => [
            bless( {
              '' => '\\left(\\cos x+j\\sin x\\right)^n=\\cos(nx)+j\\sin(nx)
  ',
              'Left' => bless( {
                'Component' => bless( {
                  '' => '\\cos x+j\\sin x',
                  'Left' => bless( {
                    'Component' => bless( {
                      'Argument' => bless( {
                        'Component' => bless( {
                          'Alphabet' => 'x',
                          'trailingws' => ''
                        }, 'SpeL::Object::Variable' ),
                        'Rest' => bless( {
                          'Op' => bless( {
                            'Op' => '+'
                          }, 'SpeL::Object::Binop' ),
                          'Remainder' => bless( {
                            'Component' => bless( {
                              'Alphabet' => 'j',
                              'trailingws' => ''
                            }, 'SpeL::Object::Variable' ),
                            'Rest' => bless( {
                              'Op' => bless( {
                                'Op' => ''
                              }, 'SpeL::Object::Binop' ),
                              'Remainder' => bless( {
                                'Component' => bless( {
                                  'Argument' => bless( {
                                    'Component' => bless( {
                                      'Alphabet' => 'x',
                                      'trailingws' => ''
                                    }, 'SpeL::Object::Variable' ),
                                    'Rest' => bless( {
                                      'ws' => ''
                                    }, 'SpeL::Object::Expressionrest' )
                                  }, 'SpeL::Object::Expression' ),
                                  'Trig' => '\\sin'
                                }, 'SpeL::Object::Function' ),
                                'Rest' => bless( {
                                  'ws' => ''
                                }, 'SpeL::Object::Expressionrest' )
                              }, 'SpeL::Object::Expression' )
                            }, 'SpeL::Object::Expressionrest' )
                          }, 'SpeL::Object::Expression' )
                        }, 'SpeL::Object::Expressionrest' )
                      }, 'SpeL::Object::Expression' ),
                      'Trig' => '\\cos'
                    }, 'SpeL::Object::Function' ),
                    'Rest' => bless( {
                      'ws' => ''
                    }, 'SpeL::Object::Expressionrest' )
                  }, 'SpeL::Object::Expression' )
                }, 'SpeL::Object::Relation' ),
                'Rest' => bless( {
                  'Op' => bless( {
                    'Lit' => 'n',
                    'sup' => '^'
                  }, 'SpeL::Object::Power' ),
                  'Remainder' => bless( {
                    'ws' => ''
                  }, 'SpeL::Object::Expressionrest' )
                }, 'SpeL::Object::Expressionrest' ),
                'cbracket' => '\\right)',
                'obracket' => '\\left('
              }, 'SpeL::Object::Expression' ),
              'RelationChain' => [
                {
                  'RelOperator' => bless( {
                    '' => '='
                  }, 'SpeL::Object::RelOperator' ),
                  'Right' => bless( {
                    'Component' => bless( {
                      'Argument' => bless( {
                        'Component' => bless( {
                          'Alphabet' => 'n',
                          'trailingws' => ''
                        }, 'SpeL::Object::Variable' ),
                        'Rest' => bless( {
                          'Op' => bless( {
                            'Op' => ''
                          }, 'SpeL::Object::Binop' ),
                          'Remainder' => bless( {
                            'Component' => bless( {
                              'Alphabet' => 'x',
                              'trailingws' => ''
                            }, 'SpeL::Object::Variable' ),
                            'Rest' => bless( {
                              'ws' => ''
                            }, 'SpeL::Object::Expressionrest' )
                          }, 'SpeL::Object::Expression' )
                        }, 'SpeL::Object::Expressionrest' )
                      }, 'SpeL::Object::Expression' ),
                      'Trig' => '\\cos',
                      'cbracket' => ')',
                      'obracket' => '('
                    }, 'SpeL::Object::Function' ),
                    'Rest' => bless( {
                      'Op' => bless( {
                        'Op' => '+'
                      }, 'SpeL::Object::Binop' ),
                      'Remainder' => bless( {
                        'Component' => bless( {
                          'Alphabet' => 'j',
                          'trailingws' => ''
                        }, 'SpeL::Object::Variable' ),
                        'Rest' => bless( {
                          'Op' => bless( {
                            'Op' => ''
                          }, 'SpeL::Object::Binop' ),
                          'Remainder' => bless( {
                            'Component' => bless( {
                              'Argument' => bless( {
                                'Component' => bless( {
                                  'Alphabet' => 'n',
                                  'trailingws' => ''
                                }, 'SpeL::Object::Variable' ),
                                'Rest' => bless( {
                                  'Op' => bless( {
                                    'Op' => ''
                                  }, 'SpeL::Object::Binop' ),
                                  'Remainder' => bless( {
                                    'Component' => bless( {
                                      'Alphabet' => 'x',
                                      'trailingws' => ''
                                    }, 'SpeL::Object::Variable' ),
                                    'Rest' => bless( {
                                      'ws' => ''
                                    }, 'SpeL::Object::Expressionrest' )
                                  }, 'SpeL::Object::Expression' )
                                }, 'SpeL::Object::Expressionrest' )
                              }, 'SpeL::Object::Expression' ),
                              'Trig' => '\\sin',
                              'cbracket' => ')',
                              'obracket' => '('
                            }, 'SpeL::Object::Function' ),
                            'Rest' => bless( {
                              'ws' => ''
                            }, 'SpeL::Object::Expressionrest' )
                          }, 'SpeL::Object::Expression' )
                        }, 'SpeL::Object::Expressionrest' )
                      }, 'SpeL::Object::Expression' )
                    }, 'SpeL::Object::Expressionrest' )
                  }, 'SpeL::Object::Expression' )
                }
              ]
            }, 'SpeL::Object::Relation' )
          ],
          'mathtag' => 'equation',
          'trailingws' => '
    '
        }, 'SpeL::Object::MathEnvironment' ),
        bless( {
          '' => '
'
        }, 'SpeL::Object::TokenSequence' )
      ]
    }, 'SpeL::Object::ElementList' )
  }, 'SpeL::Object::Document' )
}, 'SpeL::Parser::Chunk' );

