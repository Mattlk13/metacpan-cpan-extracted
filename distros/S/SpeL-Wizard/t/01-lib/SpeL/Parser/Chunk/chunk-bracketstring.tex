  \begin{equation}
    \left(\cos x+j\sin x\right)^n=\cos(nx)+j\sin(nx)
  \end{equation}
