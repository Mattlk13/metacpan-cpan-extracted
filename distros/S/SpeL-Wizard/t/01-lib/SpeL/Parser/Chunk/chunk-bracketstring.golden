$tex = bless( {
  'path' => 't/01-lib/SpeL/Parser/Chunk',
  'tree' => bless( {
    'ElementList' => bless( {
      'Element' => [
        bless( {
          'MathUnit' => [
            bless( {
              '' => '\\left(\\cos x+j\\sin x\\right)^n=\\cos(nx)+j\\sin(nx)
  ',
              'Left' => bless( {
                'Component' => bless( {
                  'Component' => bless( {
                    '' => '\\cos x+j\\sin x',
                    'Left' => bless( {
                      'Component' => bless( {
                        'Argument' => bless( {
                          'Component' => bless( {
                            'Alphabet' => 'x',
                            'trailingws' => ''
                          }, 'SpeL::Object::Variable' ),
                          'Rest' => bless( {
                            'Op' => bless( {
                              'Op' => '+'
                            }, 'SpeL::Object::Binop' ),
                            'Remainder' => bless( {
                              'Component' => bless( {
                                'Alphabet' => 'j',
                                'trailingws' => ''
                              }, 'SpeL::Object::Variable' ),
                              'Rest' => bless( {
                                'Op' => bless( {
                                  'Op' => ''
                                }, 'SpeL::Object::Binop' ),
                                'Remainder' => bless( {
                                  'Component' => bless( {
                                    'Argument' => bless( {
                                      'Component' => bless( {
                                        'Alphabet' => 'x',
                                        'trailingws' => ''
                                      }, 'SpeL::Object::Variable' ),
                                      'Rest' => bless( {
                                        'ws' => ''
                                      }, 'SpeL::Object::Expressionrest' )
                                    }, 'SpeL::Object::Expression' ),
                                    'Name' => bless( {
                                      'Op' => 'sin'
                                    }, 'SpeL::Object::Trig' )
                                  }, 'SpeL::Object::Function' ),
                                  'Rest' => bless( {
                                    'ws' => ''
                                  }, 'SpeL::Object::Expressionrest' )
                                }, 'SpeL::Object::Expression' )
                              }, 'SpeL::Object::Expressionrest' )
                            }, 'SpeL::Object::Expression' )
                          }, 'SpeL::Object::Expressionrest' )
                        }, 'SpeL::Object::Expression' ),
                        'Name' => bless( {
                          'Op' => 'cos'
                        }, 'SpeL::Object::Trig' )
                      }, 'SpeL::Object::Function' ),
                      'Rest' => bless( {
                        'ws' => ''
                      }, 'SpeL::Object::Expressionrest' )
                    }, 'SpeL::Object::Expression' )
                  }, 'SpeL::Object::Relation' ),
                  'cbracket' => '\\right)',
                  'obracket' => '\\left('
                }, 'SpeL::Object::Bracketconstruction' ),
                'Rest' => bless( {
                  'Op' => bless( {
                    'Lit' => 'n',
                    'sup' => '^'
                  }, 'SpeL::Object::Power' ),
                  'Remainder' => bless( {
                    'ws' => ''
                  }, 'SpeL::Object::Expressionrest' )
                }, 'SpeL::Object::Expressionrest' )
              }, 'SpeL::Object::Expression' ),
              'RelationChain' => [
                {
                  'RelOperator' => bless( {
                    '' => '='
                  }, 'SpeL::Object::RelOperator' ),
                  'Right' => bless( {
                    'Component' => bless( {
                      'Argument' => bless( {
                        'Component' => bless( {
                          '' => 'nx',
                          'Left' => bless( {
                            'Component' => bless( {
                              'Alphabet' => 'n',
                              'trailingws' => ''
                            }, 'SpeL::Object::Variable' ),
                            'Rest' => bless( {
                              'Op' => bless( {
                                'Op' => ''
                              }, 'SpeL::Object::Binop' ),
                              'Remainder' => bless( {
                                'Component' => bless( {
                                  'Alphabet' => 'x',
                                  'trailingws' => ''
                                }, 'SpeL::Object::Variable' ),
                                'Rest' => bless( {
                                  'ws' => ''
                                }, 'SpeL::Object::Expressionrest' )
                              }, 'SpeL::Object::Expression' )
                            }, 'SpeL::Object::Expressionrest' )
                          }, 'SpeL::Object::Expression' )
                        }, 'SpeL::Object::Relation' ),
                        'cbracket' => ')',
                        'obracket' => '('
                      }, 'SpeL::Object::Bracketconstruction' ),
                      'Name' => bless( {
                        'Op' => 'cos'
                      }, 'SpeL::Object::Trig' )
                    }, 'SpeL::Object::Function' ),
                    'Rest' => bless( {
                      'Op' => bless( {
                        'Op' => '+'
                      }, 'SpeL::Object::Binop' ),
                      'Remainder' => bless( {
                        'Component' => bless( {
                          'Alphabet' => 'j',
                          'trailingws' => ''
                        }, 'SpeL::Object::Variable' ),
                        'Rest' => bless( {
                          'Op' => bless( {
                            'Op' => ''
                          }, 'SpeL::Object::Binop' ),
                          'Remainder' => bless( {
                            'Component' => bless( {
                              'Argument' => bless( {
                                'Component' => bless( {
                                  '' => 'nx',
                                  'Left' => bless( {
                                    'Component' => bless( {
                                      'Alphabet' => 'n',
                                      'trailingws' => ''
                                    }, 'SpeL::Object::Variable' ),
                                    'Rest' => bless( {
                                      'Op' => bless( {
                                        'Op' => ''
                                      }, 'SpeL::Object::Binop' ),
                                      'Remainder' => bless( {
                                        'Component' => bless( {
                                          'Alphabet' => 'x',
                                          'trailingws' => ''
                                        }, 'SpeL::Object::Variable' ),
                                        'Rest' => bless( {
                                          'ws' => ''
                                        }, 'SpeL::Object::Expressionrest' )
                                      }, 'SpeL::Object::Expression' )
                                    }, 'SpeL::Object::Expressionrest' )
                                  }, 'SpeL::Object::Expression' )
                                }, 'SpeL::Object::Relation' ),
                                'cbracket' => ')',
                                'obracket' => '('
                              }, 'SpeL::Object::Bracketconstruction' ),
                              'Name' => bless( {
                                'Op' => 'sin'
                              }, 'SpeL::Object::Trig' )
                            }, 'SpeL::Object::Function' ),
                            'Rest' => bless( {
                              'ws' => ''
                            }, 'SpeL::Object::Expressionrest' )
                          }, 'SpeL::Object::Expression' )
                        }, 'SpeL::Object::Expressionrest' )
                      }, 'SpeL::Object::Expression' )
                    }, 'SpeL::Object::Expressionrest' )
                  }, 'SpeL::Object::Expression' )
                }
              ]
            }, 'SpeL::Object::Relation' )
          ],
          'mathtag' => 'equation',
          'trailingws' => '
    '
        }, 'SpeL::Object::MathEnvironment' ),
        bless( {
          '' => '
'
        }, 'SpeL::Object::TokenSequence' )
      ]
    }, 'SpeL::Object::ElementList' )
  }, 'SpeL::Object::Document' )
}, 'SpeL::Parser::Chunk' );

